crates/svm-codec/examples/test.js

```diff
@@ -0,0 +1,232 @@
+'use strict';
+
+const assert = require('assert');
+const fs = require('fs');
+
+const OK_MARKER = 1;
+const ERR_MARKER = 0;
+
+async function compileWasmCodec () {
+    const wasm = await WebAssembly.compile(fs.readFileSync('svm_codec.wasm'));
+    const importObject = {};
+
+    return WebAssembly.instantiate(wasm, importObject)
+}
+
+function instanceCall(instance, func_name, buf) {
+    const func = instance.exports[func_name];
+    return func(buf)
+}
+
+function wasmNewBuffer(instance, object) {
+    const objectStr = JSON.stringify(object);
+    const bytes = new TextEncoder('utf-8').encode(objectStr);
```

Comment:
All tests are failing for me locally with `ReferenceError: TextEncoder is not defined`. 

Which Nodejs version do you use? it should be specified in `package.json`. 
Mine:
```
Moshes-MacBook-Pro:examples moshababo$ node -v
v10.15.3
```

If this is problematic, maybe you should use the well-supported `Buffer` class instead (assuming this code is intended to run in Nodejs and not in a browser).

Reply:
```
❯ node -v
v12.6.0
```

Reply:
I'll add to `package.json`